js/indicators/aroon.src.js

```diff
@@ -0,0 +1,181 @@
+             * @product highstock
+             */
+            period: 14
+        },
+        marker: {
+            enabled: false
+        },
+        tooltip: {
+            pointFormat: '<span style="color:{point.color}">\u25CF</span><b> {series.name}</b><br/>Aroon Up: {point.y}<br/>Aroon Down: {point.aroonDown}<br/>'
+        },
+        /**
+         * aroonDown line options.
+         *
+         * @since 7.0.0
+         * @product highstock
+         */
+        aroonDown: {
+            /**
+             * Styles for an aroonDown line.
+             *
+             * @since 7.0.0
+             * @product highstock
+             */
+            styles: {
+                /**
+                 * Pixel width of the line.
+                 *
+                 * @type {Number}
+                 * @since 7.0.0
+                 * @product highstock
+                 */
+                lineWidth: 1,
+                /**
+                 * Color of the line. If not set, it's inherited from
+                 * [plotOptions.aroon.color](
+                 * #plotOptions.aroon.color).
+                 *
+                 * @type {Highcharts.ColorString}
+                 * @since 7.0.0
+                 * @product highstock
+                 */
+                lineColor: undefined
+            }
+        },
+        dataGrouping: {
+            approximation: 'averages'
+        }
+    }, /** @lends Highcharts.Series.prototype */ H.merge(multipleLinesMixin, {
+        nameComponents: ['period'],
+        nameBase: 'Aroon',
+        pointArrayMap: ['y', 'aroonDown'],
+        pointValKey: 'y',
+        linesApiNames: ['aroonDown'],
+        getValues: function (series, params) {
+            var period = params.period,
+                xVal = series.xData,
+                yVal = series.yData,
+                yValLen = yVal ? yVal.length : 0,
+                AR = [], // 0- date, 1- Aroon Up, 2- Aroon Down
+                xData = [],
+                yData = [],
+                slicedY,
+                low = 2,
+                high = 1,
+                aroonUp,
+                aroonDown,
+                xLow,
+                xHigh,
+                i;
+
+            // For a N-period, we start from N-1 point, to calculate Nth point
+            // That is why we later need to comprehend slice() elements list
+            // with (+1)
+            for (i = period - 1; i < yValLen; i++) {
+                slicedY = yVal.slice(i - period + 1, i + 2);
+
+                xLow = minIndexInArray(slicedY.map(function (elem) {
+                    if (elem[low]) {
+                        return elem[low];
+                    }
+                    return elem;
+                }));
+
+                xHigh = maxIndexInArray(slicedY.map(function (elem) {
+                    if (elem[high]) {
+                        return elem[high];
+                    }
+                    return elem;
+                }));
+
```

Comment:
Consider to optymize this both functions which seems to me more or less the same.